### PR TITLE
WIP: Comments from Vinesh

### PR DESCRIPTION
**Pull request by dfm**
I'm opening a pull request with this to make it easier to discuss.

### CODE REVIEW

File: documents/ms.tex [hunk 1]
```diff
@@ -258,24 +259,29 @@ \section{Introduction}
 \begin{equation}
 \label{eq:QP}
 k_{i,j} = A \exp \left[-\frac{(x_i - x_j)^2}{2l^2} -
-\Gamma^2 \sin^2\left(\frac{2\pi}{P}\right) \right].
+\Gamma^2 \sin^2\left(\frac{\vrcomment{\pi(x_i - x_j)}}{P}\right) \right].
 \end{equation}
-It is the product of the SE kernel function, which describes the overall
+\vrcomment{[VR: there was a term missing from the above definition, viz.\ $(x_i - x_j)$ inside the $\sin()$. Without it, the $\sin()$ term reduces to a constant. Also, I changed $2\pi$ to $\pi$: if you define it with $2\pi$ then the period will differ from the physical period by a factor of $2$, given the squared term modifying the sine wave: ${\sin ^2}(\frac{\theta }{2}) = \frac{1}{2} - \frac{1}{2}\cos \theta $. This can be checked by plotting some covariance kernels with large $l$ and noting that peaks appear with expecting spacing $P$ only if you remove the factor of $2$ that originally appeared in the definition.]}   It is the product of the SE kernel function, which describes the overall
```
Comment (dfm) [inline documents/ms.tex]: Ruth is using george and the kernel is implemented the way you say there so this is just a typo in the paper and it won't affect the results.

File: documents/ms.tex [hunk 2]
```diff
@@ -446,6 +461,25 @@ \subsubsection{Non-period Hyperparameters}
...
...tutorial paper and I know I've worked through the mathematics myself -- offhand I can't remember the solution, but it's a fairly straightforward looking thing involving the precision matrix like $A\sim{{\mathbf{y}}^{\textrm{T}}}{{\mathbf{K}}^{ - 1}}{\mathbf{y}}$ with some scaling constant. I'll see if I can dig up my workings and/or find a reference. I think this means that if you're anyway going to use a flat prior on $A$, your MAP solution will end up corresponding to the ML solution. In which case you could speed things up a good deal by just setting $A$ to its optimal value \emph{ab initio}, and not bothering to explore the full posterior PDF. More importantly though, if for some reason the MAP value you use doesn't correspond more or less to the ML value (e.g.\ if something went wrong with the MCMC convergence) -- in other words, if you end up using a non-optimal value for $A$ -- I've found you can sometimes get rather bizarre behaviour. For example, when I fit data with an obvious period, sometimes I end up recovering a very wrong period if $A$ is also very wrong e.g.\ much too large. Such problems can be avoided quite straightforwardly by constraining $A$ to sensible values.
```
Comment (dfm) [inline documents/ms.tex]: This must be the ML value of A _conditioned on_ the other parameters though. There's no reason why the ML value of A for some values of the other hyperparameters will be the MAP of the full posterior! I agree that you could use this to remove a parameter from the inference but I'm not convinced that that would make a huge difference.

File: documents/ms.tex [hunk 3]
```diff
@@ -446,6 +461,25 @@ \subsubsection{Non-period Hyperparameters}
...
+\item Suzanne and I have shown that provided
+\begin{equation}
+\frac{{{P^2}}}{{{l^2}{\Gamma ^2}}} < \frac{{4\pi }}{3},
+\end{equation}
+the QP covariance function will have at least one inflection point other than at $|x_i-x_j|=0$ (the derivation is straightforward though a bit long). So $\frac{{{P^2}}}{{{l^2}{\Gamma ^2}}} = \frac{{4\pi }}{3}$ corresponds to a critical point at which secondary maxima disappear, and the behaviour of the kernel becomes dominated by the SE term rather than the periodic term. For $\Gamma\sim\mathcal{O}(1)$, this translates into $l\gtrsim P/2$. In practice satisfying this criterion doesn't absolutely guarantee that function draws will have clear evidence of periodicity, but it does give you a quantitative boundary between obviously periodic behaviour and not-so-obviously periodic behaviour (if you're trying to fit rotation periods then obviously you don't want to use an SE-like kernel to do so) and provides some justification for a more stringent criterion such as $l>P$, or $l>2P$ (ensure that functions evolve significantly over time-scales longer than 2 periods), etc.
```
Comment (dfm) [inline documents/ms.tex]: 👍

File: documents/ms.tex [hunk 4]
```diff
@@ -580,11 +614,11 @@ \subsection{Sampling}
 We initialize 500 walkers with random samples from the prior and use a
 weighted mixture of the KDE, DE, and snooker proposals with weights of 0.4,
 0.4, and 0.2 respectively.
-We run fifty steps of the sampler at a time, checking for convergence after
-each iteration, up to a maximum of fifty iterations.
+We run \vrcomment{50} steps of the sampler at a time, checking for convergence after
+each iteration, up to a maximum of \vrcomment{50} iterations.
```
Comment (RuthAngus) [inline documents/ms.tex]: @timothydmorton I think Vinesh wants you to check these numbers?

**Pull request merged**